integration/kubernetes/k8s-credentials-secrets.bats

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
+
+setup() {
+	export KUBECONFIG=/etc/kubernetes/admin.conf
+	pod_config_dir="${BATS_TEST_DIRNAME}/untrusted_workloads"
+}
+
+@test "Credentials using secrets" {
+	secret_name="test-secret"
+	pod_name="secret-test-pod"
+	second_pod_name="secret-envars-test-pod"
+
+	# Create the secret
+	sudo -E kubectl create -f "${pod_config_dir}/inyect_secret.yaml"
+
+	# View information about the secret
+	sudo -E kubectl get secret "${secret_name}" -o yaml | grep "type: Opaque"
+
+	# Create a pod that has acces to the secret through a volume
+	sudo -E kubectl create -f "${pod_config_dir}/pod-secret.yaml"
+
+	# Check pod creation
+	sudo -E kubectl wait --for=condition=Ready pod "$pod_name"
+
+	# List the files
+	cmd="ls /tmp/secret-volume"
+	sudo -E kubectl exec $pod_name -- sh -c "$cmd" | grep -E "password|username"
```

Comment:
I don't really understand why alternation is being used here and below in the `grep` call? Shouldn't we be checking for both username *and* password (rather than accepting either)?

Reply:
Yes, but that `grep` will succeed in **all** the following scenarios:

1. if it **only** finds `password`.
1. if it **only** finds `username`.
1. if it find `username` **and** `password`.

But that's not what we want aiui - we need to guarantee point (3) above happens right?